src/main/scala/tech/sourced/gemini/FileQuery.scala

```diff
@@ -186,18 +185,26 @@ class FileQuery(
 
   protected def readDocFreqFromDB(): Option[OrderedDocFreq] = {
     log.info(s"Reading docFreq from DB")
-    val cols = tables.docFreqCols
-    val row = conn.execute(s"SELECT * FROM ${tables.docFreq} WHERE ${cols.id} = '${mode}'").one()
-    if (row == null) {
+    val docsCols = tables.featuresDocsCols
+    val freqCols = tables.featuresFreqCols
+    val docsRow = conn.execute(s"SELECT * FROM ${tables.featuresDocs} WHERE ${docsCols.id} = '$mode'").one()
+    if (docsRow == null) {
       log.warn("Document frequency table is empty.")
       None
     } else {
-      val df = row
-        .getMap("df", classOf[java.lang.String], classOf[java.lang.Integer])
+      var tokens = IndexedSeq[String]()
+      val df = conn
+        .execute(s"SELECT * FROM ${tables.featuresFreq} WHERE ${freqCols.id} = '$mode'")
         .asScala
-        .mapValues(_.toInt)
+        .map { row =>
+          // tokens have to be sorted, df.keys isn't sorted
```

Comment:
Alphabetical. DB already returns it in the correct order.
This map is quite big. On a small 1k repos dataset serialized version of this map took 45mb. I would prefer to avoid unnecessary sorting if it's possible.

Reply:
I don't know cassandra actually, so correct me if I'm wrong.

Isn't relying on the db ordering without actually specifying it at query time a bit dangerous? Let me explain better. I guess that in the table there's some sort of index that is being used as ordering key, but if we change something ad db level we also need to remember to change the code. Isn't it better to also include the ordering in the query? I guess that it won't affect the execution time given that even without including it still orders them using that order.

Reply:
If then change something at db level everything should continue to work. It could just perform much slower (in case for example we remove the ordering key).

Reply:
(mode, feature) is a primary key that's why it's sorted.

Thanks for pointing that I missed order by in SELECT! I'll add it.

Reply:
fixed